Jenkinsfile.cloud

```diff
@@ -69,4 +73,26 @@ node(env.NODE) {
             """
         }
     }
+
+    stage("Create AMI") {
+        docker.image(DOCKER_IMG).inside(DOCKER_ARGS) {
+            // not packaged yet, just build from src for now
+            sh "dnf install -y git golang"
+            sh "git clone https://github.com/coreos/mantle"
+            sh "mantle/build ore"
+
+            withCredentials([[$class: 'AmazonWebServicesCredentialsBinding',
+                              credentialsId: env['AWS_CREDENTIALS']]]) {
+                sh """
+                    mantle/bin/ore aws upload --region us-east-1 \
+                        --ami-name 'rhcos_dev_${commit[0..6]}' \
+                        --ami-description 'Red Hat CoreOS ${commit}' \
+                        --bucket 's3://${env.S3_PRIVATE_BUCKET}/rhcos' \
+                        --delete-object --force \
```

Comment:
`--delete-object` is the default and can be omitted.

`--force` is slightly misdocumented.  Without it, the existing S3 object will be reused (and then deleted after the snapshot is created).  With it, the existing object is clobbered, which could race with another build using the same object.  Preferably there's a 1:1 mapping between image bits and filename (so reusing an object from a previously aborted run is safe) and force is not set.

Along those lines, there's a bug in ore and a bug here.  The bug in ore is that `--bucket` is supposed to be a prefix with the basename of `--file` appended, but instead we're using `--bucket` as the complete object name.  The bug here is that `--bucket + basename(--file)` is always the same value, so if there are two simultaneous runs, one will clobber the other.

Reply:
ore bugfix at https://github.com/coreos/mantle/pull/854.

Reply:
Thanks for the review! I dropped `--force` and based the filename on the commit. Also added a tweak to the TDL to lower the virtual disk size.

Reply:
> `--delete-object` is the default and can be omitted.

Yeah, I pass it just to be more explicit. I added a comment for that.